Yona/Yona/Settings/SettingsViewController.swift

```diff
@@ -110,15 +117,22 @@ extension SettingsViewController:UITableViewDelegate {
     }
     
     func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
-        if indexPath.row == 2 {
+        if indexPath.row == 0 {
+            //change pin
+            if let login = R.storyboard.login.loginStoryboard {
+                login.isFromSettings = true
+                login.hidesBottomBarWhenPushed = true
+                self.navigationController?.pushViewController(login, animated: false)
+            }
+        } else if indexPath.row == 1 {
+            //privacy
+        } else if indexPath.row == 2 {
             callAddDeviceMethod()
-        }
-        else if indexPath.row == 3 {
+        } else if indexPath.row == 3 {
             self.displayAlertOption(NSLocalizedString("delete-user", comment: ""),cancelButton: true, alertDescription: NSLocalizedString("deleteusermessage", comment: ""), onCompletion: { (buttonPressed) in
                 switch buttonPressed {
                 case alertButtonType.OK:
                     self.callUnSubscribeMethod()
-                    
                 case alertButtonType.cancel:
                     break
                     //do nothing or send back to start of signup?
```

Comment:
What does the different indexPath.row numbers supposed to mean?
The indexes are meaningless in such situations, you can't guess the functionality by looking at the numbers. There are several ways to deal with such thing, but since it seems there are only 3 cases, we can handle them like that
`let setting = settingsArray[indexPath.row]

        if setting == NSLocalizedString("change-pin", comment: ""){
            
        }else if setting ==  NSLocalizedString("privacy", comment: ""){
            
        }else if setting == NSLocalizedString("delete-user", comment: ""){
            
        }`

At least it now contains words instead of numbers, which can help you:
1- Easily guess what this code supposed to do
2- It is not depending on the order any more.